django/thunderstore/api/cyberstorm/views/package_detail.py

```diff
@@ -140,6 +148,11 @@ def get_custom_package_detail_listing(
                     ratings=Count("package__package_ratings"),
                 ).values("ratings"),
             ),
+            has_changelog=Case(
+                When(package__latest__changelog__regex=r"[^\s]", then=True),
```

Comment:
Do we really care about filtering out empty changelogs? IMO if someone submits a changelog that's only whitespace or entirely empty we shouldn't pretend like it doesn't exist at all. After all, we have a clear signal for a changelog existing, which is that the file exists in the package. If it doesn't, the db field is set to null.

Generally I'd prefer keeping the rendering/consumer side dumb and validation smart. The philosophy behind this is that there's likely going to be many more implementations for consuming packages than there are implementations validating them, and ultimately we can enforce the rules on the validation level much more efficiently than on the consumer side.

Another aspect to this is that if we filter out empty changelogs when rendering, there's no pressure on package creators to care about formatting theirs properly or deleting it if it doesn't contain anything. In other words if we implement this _without_ also adding validation, it somewhat enables bad behavior in the ecosystem. On the other hand if we add validation, this becomes unnecessary to handle on the rendering side.

Reply:
That's fair, I guess I was working with wrong passive assumption about when the field can be an empty string.